Cpp_hands_on_programming/lib/queue/Ipc_queue.cc

```diff
@@ -9,103 +9,103 @@ void Ipc_queue::send()
-    //         if (mq_send(msg_queue, buffer.data(), 0, 0) == -1) {
-    //             throw(std::runtime_error("Ipc_queue::send: mq_send 0 byte message"));
-    //         }
-    //     }
-    // }
+        total += bytes;
+        if (total == size) {
+            if (mq_send(msg_queue, buffer.data(), 0, 0) == -1) {
+                throw(std::runtime_error("Ipc_queue::send: mq_send 0 byte message"));
+            }
+        }
+    }
 
-    // if (total == size) {
-    //     std::cout << "File delivered by queue successfully, exiting the program.."  << std::endl;
-    // } else {
-    //     throw(std::runtime_error("Ipc_queue::send: total bytes sent and file size different"));
-    // }
+    if (total == size) {
+        std::cout << "File delivered by queue successfully, exiting the program.."  << std::endl;
+    } else {
+        throw(std::runtime_error("Ipc_queue::send: total bytes sent and file size different"));
+    }
 }
 
 void Ipc_queue::receive()
 {
     std::cout << "Starting queueReceive.." << std::endl;
-    // check_leadingSlash();
+    check_Slash();
 
-    // std::vector<char> buffer;
-    // int total(0), bytes, size(0);
-    // // open msg queue loop
-	// while ((msg_queue = mq_open(name.c_str(), O_RDONLY)) == -1) {
-	// 	std::this_thread::sleep_for (std::chrono::seconds(1));
-	// }
+    std::vector<char> buffer;
+    int total(0), bytes, size(0);
+    // open msg queue loop
+	while ((msg_queue = mq_open(name.c_str(), O_RDONLY)) == -1) {
+		std::this_thread::sleep_for (std::chrono::seconds(1));
+	}
     
-    // // get queue attr 
-    // mq_getattr(msg_queue, &attr);
-    // // initialize buffer size
-    // buffer.resize(attr.mq_msgsize);
+    // get queue attr 
+    mq_getattr(msg_queue, &attr);
+    // initialize buffer size
+    buffer.resize(attr.mq_msgsize);
 
-    // // Start receive write loop 
-    // do {
-    //     if ((bytes = mq_receive(msg_queue, buffer.data(), attr.mq_msgsize, 0)) == -1) {
-    //         throw(std::runtime_error("Ipc_queue::receive: mq_receive"));
-    //     }
-    //     if (writeFile(filename, total, buffer, bytes) == -1) {
-    //         throw(std::runtime_error("Ipc_queue::receive: writeFile"));
-    //     }
-    //     total += bytes;
-    // } while (bytes != 0);
-    // // get newly written file size
-    // if ((size = getFileSize(filename)) == -1) {
-	// 	throw(std::runtime_error("Ipc_method:: getFileSize"));
-	// }
-    // if (size == total) {
-    //     std::cout << "Write to file from queue successfully, exiting the program.."  << std::endl;
-    // } else {
-    //     throw(std::runtime_error("Ipc_queue::receive: File sizes difference between sent file and receive file"));
-    // }
+    // Start receive write loop 
+    do {
+        if ((bytes = mq_receive(msg_queue, buffer.data(), attr.mq_msgsize, 0)) == -1) {
+            throw(std::runtime_error("Ipc_queue::receive: mq_receive"));
+        }
+        if (writeFile(filename, total, buffer, bytes) == -1) {
```

Comment:
I think there is redundancy in 'writeFile' arguments. vector already contains info about number of bytes (just needs to be resized)